databases/recutils/Portfile

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                recutils
+version             1.8
+
+categories          databases
+maintainers         @zaid openmaintainer
+description         Tools to work with human-editable, plain text data files
+long_description    \
+    GNU Recutils is a set of tools and libraries to access human-editable, \
+    plain text databases called recfiles. The data is stored as a sequence \
+    of records, each record containing an arbitrary number of named fields.
+
+platforms           darwin
+license             GPL-3+
+
+homepage            https://www.gnu.org/software/recutils/
+master_sites        https://ftp.gnu.org/gnu/recutils/
+mirror              https://ftpmirror.gnu.org/gnu/recutils/
+checksums           sha256  df8eae69593fdba53e264cbf4b2307dfb82120c09b6fab23e2dad51a89a5b193 \
+                    rmd160  d3f0042a5898ad5630f1295d9fa2f87a08031a34 \
+                    size    2474024
+
+depends_lib         port:libgcrypt \
+                    port:ossp-uuid \
+                    port:flex \
+                    port:check
+
+configure.args      --disable-dependency-tracking
+configure.cflags-append -Wno-error=implicit-function-declaration
```

Comment:
@reneeotten good point, I didn't like having to add this either. 

I think the compiler defaults changed in recent versions because that warning is treated as an error on MacOS 11.x.

I've been doing the work on MacOS 10.13 so the build succeeded with warnings but doing the same thing on 11.5 fails due to `implicit-function-declaration`.

Reply:
@reneeotten I heard back from upstream about the warnings:

>These implicit function declarations happen in code generated by flex.
I would suggest you to pass -Wno-implicit-function-declaration in CFLAGS
in your build script or recipe.

Full reply can be found [here](https://lists.gnu.org/archive/html/bug-recutils/2021-09/msg00001.html)

Reply:
@reneeotten I'm going to re-add the `-Wno-error=implicit-function-declaration` CFLAG unless you have other suggestions for me to try.

Reply:
sorry for the delay @zaid .... that's not a very helpful reply unfortunately. It could be that it happens in the code generated by `flex`, but I'd assume that this is still caused by the input file provided by the project. 

Having said that, I personally don't have any interest in this port and will not spend time to look into this. So if upstream doesn't want to fix it and it works for you like this then I suppose we could merge this - perhaps someone else within the MacPorts community (not looking at GitHub PRs) will be able to point out a fix.

Can you please add the link to your report and upstream's answer in the Portifile where you add the `-Wno-implicit-function-declaration` so that it's clear that you attempted to resolve this? Thanks again!

Reply:
No worries @reneeotten ! thank you for taking the time to review this PR 😄 

I added a comment linking to the mailing list post to the `Portfile`, please let me know if you'd like that changed / tweaked a bit more.

I verified that the updated port works on older versions of MacOS (tested on High Sierra) as well as Big Sur.

Reply:
thanks @ziad! Hopefully upstream will fix the underlying error (or perhaps someone here knows how to do that), but for now let's just merge this!